docs/devguide/dtypes.rst

```diff
@@ -1,2 +1,178 @@
 Implementing a New Data Type
 ============================
+
+COMPAS data types are classes that are based on :class:`compas.data.Data`.
+
+Data types can be serialized to JSON with
+
+* :func:`compas.json_dump`
+* :func:`compas.json_dumps`
+* :func:`compas.json_dumpz`
+
+and deserialized with the corresponding "load" functions
+
+* :func:`compas.json_load`
+* :func:`compas.json_loads`
+* :func:`compas.json_loadz`
+
+All geometry objects and data structures,
+and also, for example, the visualization scene,
+are serializable data types.
+
+
+Creating a new data type
+========================
+
+In most cases, it is sufficient to implement the ``__data__`` property when creating your custom `Data` class.
+
+.. code-block:: python
+
+    class CustomData(Data):
```

Comment:
Perhaps nitpicking, but since the word "data" is used a lot in this section, it might be good to find another name for the custom class that is not `CustomData`. Maybe even make it something less abstract, `Brick`? `Building`?

Reply:
sure but something neutral then, because people get distracted...

Reply:
Something generic like `Element`, `Entity`, `Item`, or even `ExampleClass`?
Something in the realm of fictional or theoretical geometry like `Hypercube`, `Tesseract`, `KleinBottle`, `Unobtainium`?
Something boring and business-y, like `Customer`, `Person`, `Project`, `Invoice`?
Something silly like `Unicorn`, `Pirate`, `Dinosaur`?
Something of a reference to pop culture, like `Hexahedron`, `Chronohedron`, `Holocron`, `Monolith`, `Vogon`?